packages/zoe/src/contracts/otcDeskReputation.js

```diff
@@ -0,0 +1,142 @@
+// @ts-check
+
+import { E } from '@agoric/eventual-send';
+import { assert, details } from '@agoric/assert';
+import { trade } from '../contractSupport';
+
+import '../../exported';
+
+/**
+ * This OTC Desk contract does not assume that the underlying assets
+ * have been escrowed. Rather, a reputation mechanism
+ * is used to inform potential users of how often the quote is
+ * accurate, and how often the trade fails.
+ *
+ * @type {ContractStartFn}
+ */
+const start = zcf => {
+  const { zcfSeat: marketMakerSeat } = zcf.makeEmptySeatKit();
+
+  let numTradesAttempted = 0;
+  let numTradesDefected = 0;
+
+  const makeQuote = (price, assets, timerAuthority, deadline) => {
+    let quoteActive = true;
+    E(timerAuthority).setWakeup(
+      deadline,
+      harden({ wake: () => (quoteActive = false) }),
+    );
+    const acceptQuote = seat => {
+      assert(quoteActive, details`The quote was no longer active`);
+      const { want: userWantActual, give: userGiveActual } = seat.getProposal();
+      numTradesAttempted += 1;
+      const hasInventory = Object.entries(assets).every(([keyword, amount]) => {
+        const allocatedAmount = marketMakerSeat.getAmountAllocated(
+          keyword,
+          amount.brand,
+        );
+        return zcf.getAmountMath(amount.brand).isGTE(allocatedAmount, amount);
+      });
+      if (!hasInventory) {
+        numTradesDefected += 1;
+        throw Error(
+          `The market maker did not have the inventory they promised. Their rating has been adjusted accordingly`,
+        );
+      }
+
+      // If this trade does not conserve rights, it will throw an
+      // error and the user will get their funds back.
+      try {
+        trade(
+          zcf,
+          {
+            seat: marketMakerSeat,
+            gains: price,
+            losses: assets,
+          },
+          { seat, gains: userWantActual, losses: userGiveActual },
+        );
+      } catch (err) {
+        console.log('actual proposal: ', seat.getProposal());
+        console.log('quote: ', {
+          assets,
+          price,
+        });
+        throw Error(
+          'Your offer did not match the quote. See the console log for more details.',
+        );
+      }
+      seat.exit();
+      return 'Trade successful';
+    };
+
+    const customProperties = harden({
+      underlyingAssets: assets,
+      strikePrice: price,
+      byTimerAuthority: timerAuthority,
+      beforeOrAtDeadline: deadline,
+    });
+
+    const acceptQuoteInvitation = zcf.makeInvitation(
+      acceptQuote,
+      'acceptQuote',
+      customProperties,
+    );
+    return acceptQuoteInvitation;
+  };
+
+  const addInventory = seat => {
+    // Take everything in this seat and add it to the marketMakerSeat
+    trade(
+      zcf,
+      { seat: marketMakerSeat, gains: seat.getCurrentAllocation() },
+      { seat, gains: {} },
+    );
+    seat.exit();
+    return 'Inventory added';
+  };
+
+  const removeInventory = seat => {
```

Comment:
given `depositToSeat` and the like, these might be simpler/clearer as just functions that just take and return Payments. Then we avoid one-sided invites with no offer safety.

Reply:
Then we don't use the payout mechanism, though. I think we should not have users send payments and receive payments outside of the Zoe escrow-> payout mechanism